File: lib/liveplace_web/channels/user_socket.ex
defmodule LiveplaceWeb.UserSocket do
  use Phoenix.Socket
  channel "game:*", LiveplaceWeb.GameChannel

  @impl true
  def connect(_params, socket, _connect_info) do
    {:ok, socket}
  end

  @impl true
  def id(_socket), do: nil
end

File: assets/js/user_socket.js
import { Socket } from "phoenix";

let socket = new Socket("/user-socket", { params: { token: window.userToken } });

socket.connect();

let channel = socket.channel("room:lobby", {});
channel
  .join()
  .receive("ok", (resp) => {
    console.log("Joined successfully", resp);
  })
  .receive("error", (resp) => {
    console.log("Unable to join", resp);
  });

export default socket;
